src/TypeScript/Analyzer/TypeScriptAnalyzer.cs

```diff
@@ -80,10 +72,10 @@ public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
         {
             Debug.Assert(IsAnalysisSupported(detectedLanguages));
 
-            ExecuteAnalysis(path, consumer, cancellationToken).Forget(); // fire and forget
+            ExecuteAnalysisAsync(path, consumer, cancellationToken).Forget(); // fire and forget
         }
 
-        internal async Task ExecuteAnalysis(string filePath, IIssueConsumer consumer, CancellationToken cancellationToken)
+        internal async Task ExecuteAnalysisAsync(string filePath, IIssueConsumer consumer, CancellationToken cancellationToken)
         {
             telemetryManager.LanguageAnalyzed("ts");
```

Comment:
you could move this and the following 3 lines to the base class as well. Just instead of passing analysisStatusNotifier you would pass the 2 strings ("ts", could be called `analysedLanguageTelemetryName` and nameof(TypeScriptAnalyzer), could be called `analyzerType` or something like that